### PR TITLE
Add comprehensive minifier tests

### PR DESCRIPTION
**Pull request by williamwen42**
Adds tests for https://github.com/pytorch/pytorch/issues/93673.

To run: `pytest test/dynamo/test_minifier.py`.

Actually runs minifier launcher script and repro scripts, rather than just checking for existence of the minifier launcher script.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: test/dynamo/test_minifier.py [hunk 1]
```diff
@@ -47,65 +151,445 @@ def setUp(self):
...
+torch._dynamo.config.repro_level = {repro_level}
 
-        inner()
+{run_code}
+"""
 
-        self.assertTrue(os.path.exists(repro_file))
+    # Runs a full minifier test.
+    # Minifier tests generally consist of 3 stages:
+    # 1. Run the problematic code (in a separate process since it could segfault)
```
Comment (mlazos) [inline test/dynamo/test_minifier.py:219]: Did you add tests for the segfault case? Nice!

Comment (williamwen42) [reply test/dynamo/test_minifier.py:219]: The after aot C++ tests abort (similar to segfault).

**Review by mlazos — APPROVED** [review]

**Review by anijain2305 — APPROVED** [review]
Very nice! Thanks for adding the extensive testing :)

File: test/dynamo/test_minifier.py [hunk 2]
```diff
@@ -1,27 +1,138 @@
...
+def triton_runtime_error(x):
+    return f"{x}; assert?"
+"""
+
+TRITON_ACCURACY_ERROR = """\
+def triton_accuracy_error(x):
+    return f"{x} + 1"
+"""
+
+DEBUG_DIR = "/tmp/_torchdynamo_debug_/"
```
Comment (wconstab) [inline test/dynamo/test_minifier.py:107]: I think this is breaking CI? @williamwen42

**Comment by wconstab** [comment]
@pytorchhbot revert

**Review by malfet — COMMENTED** [review]
LGTM, but please add a comment that PR explicitly skips testing on MacOS

**Pull request closed without merging**